### PR TITLE
Clean up doc/Makefile a bit

### PR DESCRIPTION
**Pull request by ulfalizer**
This series cleans up `doc/Makefile` a bit by removing some unused/probably unused stuff and some redundant targets.

Please check if the `latex_paper_size` stuff is used. I just removed it. It was removed in Sphinx 1.6.

### CODE REVIEW

**Review by mbolivar — APPROVED** [review]
Seems sane

File: doc/Makefile [hunk 1]
```diff
@@ -17,18 +12,14 @@ $(error The '$(SPHINXBUILD)' command was not found. Make sure you have Sphinx in
 endif
 
 # Internal variables.
-PAPEROPT_a4     = -D latex_paper_size=a4
-PAPEROPT_letter = -D latex_paper_size=letter
-ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) .
+allsphinxopts   = -d $(BUILDDIR)/doctrees $(SPHINXOPTS) .
```
Comment (andrewboie) [inline doc/Makefile]: why rename just this var to lowercase?

**Review by dbkinder — APPROVED** [review]
We currently only use the Sphinx html build for documentation (and not latex, manpages, help, or any of the other choices.)  Not sure these changes are needed, but I've got no objection.

File: doc/Makefile (hunk 1 above)
Comment (ulfalizer) [reply doc/Makefile]: @andrewboie 
See https://github.com/zephyrproject-rtos/zephyr/pull/8288/commits/7698de922a16b7aa35d727930528bf2b08d1d45c.

Seen that idiom in a bunch of places, and I think it makes Makefiles a lot clearer. It is mentioned in the GNU Make [manual](https://www.gnu.org/software/make/manual/html_node/Using-Variables.html) as well:

```
It is traditional to use upper case letters in variable names, but we recommend using lower case letters for variable names that serve internal purposes in the makefile, and reserving upper case for parameters that control implicit rules or for parameters that the user should override with command options (see Overriding Variables).
```

**Comment by ulfalizer** [comment]
@dbkinder 
I'm all for removing stuff that's not used as well. Didn't know only the `html` stuff was used.

**Review by nashif — APPROVED** [review]

**Pull request merged**